internal/cloudsql/instance.go

```diff
@@ -147,6 +147,18 @@ func (i *Instance) ConnectInfo(ctx context.Context) (map[string]string, *tls.Con
 	return res.md.ipAddrs, res.tlsCfg, nil
 }
 
+// ForceRefresh triggers an immediate refresh operation to be scheduled and used for future connection attempts.
+func (i *Instance) ForceRefresh() {
+	i.resultGuard.Lock()
+	defer i.resultGuard.Unlock()
+	// If the next refresh hasn't started yet, we can cancel it and start an immediate one
+	if i.next.Cancel() {
```

Comment:
Slightly tangential: I see that we're saving reference to `i.timer` only to stop it here, but we're not draining the timer's channel when stopping it. Since we're not using the channel, this isn't a big deal (I think), but still wonder: should we drain the channel in `Cancel`?